src/WebJobs.Extensions.OpenAI.AzureAISearch/AzureAISearchProvider.cs

```diff
@@ -269,32 +286,86 @@ async Task IndexDocumentsBatchAsync(SearchClient searchClient, IndexDocumentsBat
                 succeeded);
     }
 
-    SearchIndexClient GetSearchIndexClient(string endpoint)
+    SearchIndexClient GetSearchIndexClient(ConnectionInfo connectionInfo)
     {
-        string? key = this.configuration.GetValue<string>(this.searchAPIKeySetting);
-        if (string.IsNullOrEmpty(key))
-        {
-            return new SearchIndexClient(new Uri(endpoint), new DefaultAzureCredential());
-        }
-        else
-        {
-            return new SearchIndexClient(new Uri(endpoint), new AzureKeyCredential(key));
-        }
+        SearchIndexClient searchIndexClient;
+        (searchIndexClient, string endpoint) =
+        this.searchIndexClients.GetOrAdd(
+            connectionInfo.ConnectionName,
+            name =>
+            {
+                searchIndexClient = string.IsNullOrEmpty(this.apiKey) ? new(new Uri(this.endpoint),
+                this.GetSearchTokenCredential()) : new(new Uri(this.endpoint), new AzureKeyCredential(this.apiKey));
```

Comment:
nit: this code is hard to understand because of the whitespace. Formatting like the following should be preferred.
```csharp
searchIndexClient = string.IsNullOrEmpty(this.apiKey) ?
    new(new Uri(this.endpoint), this.GetSearchTokenCredential()) :
    new(new Uri(this.endpoint), new AzureKeyCredential(this.apiKey));
```